src/Graphics/OpenGLContext/GLSL/glsl_CombinerProgramBuilder.cpp

```diff
@@ -480,30 +479,15 @@ class ShaderBlender1 : public ShaderPart
 				"    lowp vec4 blend1 = (muxpm0 * muxa) + (muxpm2 * muxb);					\n"
 				"    clampedColor.rgb = clamp(blend1.rgb, 0.0, 1.0);						\n"
 				"  } else {																	\n"
+// Workaround for Intel drivers for Mac, issue #1601
+#if defined(OS_MAC_OS_X)
+				"      clampedColor.rgb = muxPM[uBlendMux1[0]].rgb;							\n"
+#else
 				"      lowp vec4 muxpm0 = muxPM * vec4(equal(vec4(uBlendMux1[0]), vprobe));	\n"
 				"      clampedColor.rgb = muxpm0.rgb;										\n"
+#endif
```

Comment:
I don't understand.
Does Intrel drivers work ok with 
	"    lowp vec4 muxpm0 = muxPM*vec4(equal(vec4(uBlendMux2[0]), vprobe));		\n"
in "  if (uForceBlendCycle2 != 0) " part of shader,
but fails on the same string in "else" part ?!
It is crazy. @clobber  Are you sure that your Mac is hapy with this version of blender shaders and all green color issues gone with it?

Reply:
Yeah I don't understand it either...some really odd bug. I'm sure that all the incorrect texture color issues are gone. I checked 20 games in multiple areas and the issue is gone. It seems @loganmc10's original code in his PR was fine until the F-Zero X fix was added in. Now everything is good though.

Reply:
However, if you have an idea of certain games I should try, let me know. Otherwise, I think it all looks good.

Reply:
Ok